instrumentation/grpc-1.6/testing/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/AbstractGrpcStreamingTest.java

```diff
@@ -264,6 +268,92 @@ public void onCompleted() {
                                                     (long) Status.Code.OK.value()))))));
   }
 
+  @Test
+  void grpcServerSpanEndsAfterChildSpan() throws Exception {
+    Tracer tracer = testing().getOpenTelemetry().getTracer("test");
+    AtomicBoolean serverSpanRecording = new AtomicBoolean();
+    CountDownLatch latch = new CountDownLatch(2);
+
+    BindableService greeter =
+        new GreeterGrpc.GreeterImplBase() {
+          @Override
+          public StreamObserver<Helloworld.Response> conversation(
+              StreamObserver<Helloworld.Response> observer) {
+            return new StreamObserver<Helloworld.Response>() {
+              Span span;
+
+              @Override
+              public void onNext(Helloworld.Response value) {
+                span = tracer.spanBuilder("child").startSpan();
+                observer.onNext(value);
+              }
+
+              @Override
+              public void onError(Throwable t) {
+                observer.onError(t);
+                span.end();
+              }
+
+              @Override
+              public void onCompleted() {
+                observer.onCompleted();
+                serverSpanRecording.set(Span.current().isRecording());
+                span.end();
+                latch.countDown();
+              }
+            };
+          }
+        };
+
+    Server server = configureServer(ServerBuilder.forPort(0).addService(greeter)).build().start();
+    ManagedChannel channel = createChannel(server);
+    closer.add(() -> channel.shutdownNow().awaitTermination(10, TimeUnit.SECONDS));
+    closer.add(() -> server.shutdownNow().awaitTermination());
+
+    GreeterGrpc.GreeterStub client = GreeterGrpc.newStub(channel).withWaitForReady();
+
+    StreamObserver<Helloworld.Response> observer2 =
+        client.conversation(
+            new StreamObserver<Helloworld.Response>() {
+              @Override
+              public void onNext(Helloworld.Response value) {}
+
+              @Override
+              public void onError(Throwable t) {}
+
+              @Override
+              public void onCompleted() {
+                latch.countDown();
+              }
+            });
+
+    Helloworld.Response message = Helloworld.Response.newBuilder().setMessage("message").build();
+    observer2.onNext(message);
+    observer2.onCompleted();
+
+    latch.await(10, TimeUnit.SECONDS);
+
+    // server span should end after child span
```

Comment:
nit: you actually want to test parent-child, right?

Reply:
I don't get it. I wanted to test whether in `onCompleted` we have a span that is recording (not ended).